core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -130,9 +130,9 @@ public Long getDeadlineNanoTime() {
    * Returns a new {@code CallOptions} with a request key for affinity-based routing.
    */
   @ExperimentalApi
-  public CallOptions withRequestKey(@Nullable RequestKey requestKey) {
+  public CallOptions withAttributes(@Nullable Attributes attributes) {
```

Comment:
It's probably safer to not allow null, and use Attributes.EMPTY as the default value.